utils/src/main/java/com/cloud/utils/ssh/SSHCmdHelper.java

```diff
@@ -77,8 +80,47 @@ public String getStdErr() {
     }
 
     public static com.trilead.ssh2.Connection acquireAuthorizedConnection(String ip, int port, String username, String password) {
+        return acquireAuthorizedConnection(ip, 22, username, password, null);
+    }
+
+    public static boolean acquireAuthorizedConnectionWithPublicKey(final com.trilead.ssh2.Connection sshConnection, final String username, final String privateKey) {
+        if (StringUtils.isNotBlank(privateKey)) {
+            File privateKeyFile = null;
+            try {
+                privateKeyFile = File.createTempFile("cloudstack-host-", null);
+                FileWriter writer = new FileWriter(privateKeyFile.getAbsolutePath());
+                writer.write(privateKey);
+                writer.close();
+            } catch (IOException e) {
+                s_logger.warn("An exception occurred when create a tmp file and write private key to the tmp file");
+                return false;
+            }
+            try {
+                if (!sshConnection.authenticateWithPublicKey(username, privateKeyFile, null)) {
```

Comment:
@weizhouapache can we pass the privatekey content to this method (or similar available methods)?

Reply:
@rhtyd yes, I will use `char[]`, testing it.

Reply:
@rhtyd @DaanHoogland 
updated this PR to use `char[]` in server authentication. thanks for your review !